src/ruby/puppet-server-lib/puppet/server/master.rb

```diff
@@ -126,14 +89,17 @@ def terminate
   private
 
   def self.getManifests(env)
-    manifests = []
-    if env.manifest != Puppet::Node::Environment::NO_MANIFEST
-      if File.directory?(env.manifest)
-        manifests = Dir.glob(File.join(env.manifest, '**/*.pp')).sort
-      else
-        manifests = [env.manifest]
+    manifests =
+      case
+        when env.manifest == Puppet::Node::Environment::NO_MANIFEST
+          []
+        when File.directory?(env.manifest)
+          Dir.glob(File.join(env.manifest, '**/*.pp'))
+        when File.exists?(env.manifest)
```

Comment:
In writing a new unit test for the `class-info-test` namespace, "non existent manifest dir for environment", I noticed that if the manifest directory setting pointed to a directory that doesn't actually exist on disk that an empty entry for it - as a "file" - would end up in the payload returned from a call to `getClassInfoForEnvironment`.  This condition eliminates that erroneous element.